### PR TITLE
feat(snql) Turn SnQL on for one referrer

### PR DESCRIPTION
**Pull request by evanh**
Start with the sessions stability sort referrer since the query is relatively
simple and it's also fairly low volume.

Depends on https://github.com/getsentry/sentry/pull/24900

### CODE REVIEW

File: mypy.ini [hunk 1]
```diff
@@ -18,6 +19,9 @@ warn_unused_ignores=True
 warn_return_any=True
 no_implicit_reexport=True
 
+; Set this to skip until more of the codebase is typed
+follow_imports = skip
```
Comment (evanh) [inline mypy.ini:23]: Since most of the codebase is untyped, following imports results in a lot of useless errors. Maybe when we switch mypy.ini to be on full modules instead of random files we can remove this.

File: src/sentry/utils/snuba.py [hunk 2]
```diff
@@ -639,6 +640,10 @@ def raw_query(
         is_grouprelease=is_grouprelease,
         **kwargs,
     )
+
+    if should_use_snql(referrer):
+        use_snql = True
```
Comment (lynnagara) [inline src/sentry/utils/snuba.py:645]: I don't quite get why we pass an argument `use_snql` into this function then discard it's value only if `should_use_snql` returns true but keep its value if false. The referrer list hardcoded in the snql module is also separate to the referrers that might actually pass `use_snql=True`, and it's unclear to me why there's two ways to make a referrer potentially use snql.

Comment (evanh) [reply src/sentry/utils/snuba.py:645]: Right now it's there for testing purposes and maybe forward compatibility.

Comment (evanh) [reply src/sentry/utils/snuba.py:645]: That makes sense. I'll change it to be more intuitive.

**Comment by lynnagara** [comment]
I'm not sure if the test that is still failing a flaky one or not, it seems unlikely to me that your code caused changes like that

**Review by lynnagara — APPROVED** [review]

**Comment by evanh** [comment]
> I'm not sure if the test that is still failing a flaky one or not

That was my fault, that test is very brittle. I forgot to check it before pushing.

File: src/sentry/options/defaults.py [hunk 3]
```diff
@@ -193,6 +193,7 @@
 register("snuba.search.max-total-chunk-time-seconds", default=30.0)
 register("snuba.search.hits-sample-size", default=100)
 register("snuba.track-outcomes-sample-rate", default=0.0)
+register("snuba.snql.referrer-rate", default=0.0)
```
Comment (fpacifici) [inline src/sentry/options/defaults.py:196]: Are we still supposed to register options in a separate PR from the one we start using it to be deploy safe?
If yes I think this should go in a separate PR and land before the rest.

File: src/sentry/notifications/manager.py [hunk 4]
```diff
@@ -24,7 +24,7 @@
 )
```
Comment (evanh) [inline src/sentry/notifications/manager.py:24]: Incidental type fixes from my mypy config change

File: src/sentry/snuba/outcomes.py [hunk 5]
```diff
@@ -95,7 +95,7 @@ def aggregation(self, dataset: Dataset) -> Tuple[str, str, str]:
             return ("count()", "", "times_seen")
```
Comment (evanh) [inline src/sentry/snuba/outcomes.py:95]: Incidental type fixes from my mypy config change

File: src/sentry/utils/codecs.py [hunk 6]
```diff
@@ -72,7 +72,7 @@ class JSONCodec(Codec[JSONData, str]):
     """
```
Comment (evanh) [inline src/sentry/utils/codecs.py:72]: Incidental type fixes from my mypy config change

File: src/sentry/utils/kvstore/cache.py [hunk 7]
```diff
@@ -27,13 +27,13 @@ def __init__(self, backend: BaseCache) -> None:
         self.backend = backend
```
Comment (evanh) [inline src/sentry/utils/kvstore/cache.py:27]: Incidental type fixes from my mypy config change

**Review by wmak — APPROVED** [review]
LGTM, dunno if we already have but might be worth letting the release health team know just in case there's unintended side effects
